File: Ahorn/entities/ELS_Termina_Boss_Intro.jl
module ELS_Termina_Boss_Intro

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/ELS_Termina_Boss_Intro" ELS_Termina_Boss_Intro(x::Integer, y::Integer, activationMode::String="touch", completionFlag::String="ch21_els_termina_boss_intro", removeAfterTrigger::Bool=true, requireFlag::String="", showSprite::Bool=false, texturePath::String="objects/Ingeste/sampleEntity/idle00")

const placements = Ahorn.PlacementDict(
    "default" => Ahorn.EntityPlacement(ELS_Termina_Boss_Intro)
)

function Ahorn.selection(entity::ELS_Termina_Boss_Intro)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ELS_Termina_Boss_Intro, room::Maple.Room)
    Ahorn.drawSprite(ctx, "objects/Ingeste/sampleEntity/idle00", entity.x, entity.y)
end

end

File: Loenn/entities/ELS_Termina_Boss_Intro.lua
local utils = require("utils")
local npcSessionFlags = require("ui.forms.fields.npc_session_flags")

local elsTerminaBossIntro = {}

elsTerminaBossIntro.name = "MaggyHelper/ELS_Termina_Boss_Intro"
elsTerminaBossIntro.depth = 0
elsTerminaBossIntro.justification = {0.5, 1.0}
elsTerminaBossIntro.texture = "objects/Ingeste/sampleEntity/idle00"

elsTerminaBossIntro.placements = {
    {
        name = "default",
        data = {
            activationMode = "touch",
            requireFlag = "",
            completionFlag = "ch21_els_termina_boss_intro",
            removeAfterTrigger = true,
            showSprite = false,
            texturePath = "objects/Ingeste/sampleEntity/idle00"
        }
    }
}

elsTerminaBossIntro.fieldInformation = {
    activationMode = {
        fieldType = "string",
        editable = false,
        options = {
            "interact",
            "touch",
            "roomEnter"
        }
    },
    requireFlag = npcSessionFlags,
    completionFlag = npcSessionFlags,
    removeAfterTrigger = {
        fieldType = "boolean"
    },
    showSprite = {
        fieldType = "boolean"
    },
    texturePath = {
        fieldType = "string"
    }
}

elsTerminaBossIntro.fieldOrder = {
    "x",
    "y",
    "activationMode",
    "requireFlag",
    "completionFlag",
    "removeAfterTrigger",
    "showSprite",
    "texturePath"
}

function elsTerminaBossIntro.texture(room, entity)
    return entity.texturePath or "objects/Ingeste/sampleEntity/idle00"
end

function elsTerminaBossIntro.selection(room, entity)
    return utils.rectangle(entity.x - 8, entity.y - 16, 16, 16)
end

return elsTerminaBossIntro
